Core/Battler.gd

```diff
@@ -70,16 +73,20 @@ func _on_enemy_start_turn() -> void:
 	# apply status
 	for enemy: Entity in _enemy_list:
 		enemy.get_status_component().apply_turn_start_status()
```

Comment:
i think `_handle_deaths() should be called after each enemy's turn start status has been applied.
Imagine the following situation:
Two enemies have 1 HP, each has a dot of poison. One enemy heals other enemies when it dies, by 3 HP, let's say enemy A
- If enemy A dies then enemy B, enemy A heals B then B gets his poison dot applies and survives
- If enemy B dies then enemy A, both die
- In the current situation, since we add all the enemies to the list of enemies to remove, both enemies die in each case because they are removed before actions take effect

I don't know if this is preferable, maybe we want to keep it as is so the order of the enemies doesn't influence situations

Reply:
A fair point, although if I wanted to call it after each enemy's turn I would need to do same thing as I did for their attacks as it is within loop through _enemy_list.

Actually, in most games I know, statuses like poison etc are only resolved when it comes to specific enemies' turn, however that would require us to rework the enemy turn to have them first be hit with status effects, then attack and only then hand the turn over to the next enemy. 
Currently, statuses as poison etc are specifically resolved as "on turn start" events, which implies they should be handled right at the start of the turn start
 
If we wanted to rework it into what i mentioned above, I am not sure how much work that would require - maybe it would want to be a separate issue
If we want to keep them as is and be resolved at the start of the turn, then I'm not sure which order would be preferred 🤔

Reply:
Maybe it helps to separate the status events into two separate cases?
`on_turn_start` = when the `ENEMY_ATTACKING` phase starts
`on_enemy_before_action` = before the specific enemy's action. Only applied to the enemy that is casting its action.

Also, #38 will make each enemy attack one after another. So Enemy A attacks, plays an animation. After some delay, Enemy B attacks and plays its animation.

Reply:
I see, I misremembered that STS does it like that then. 
The effects that would be affected by playing them in that order are mainly damage debuffs - so that if an enemy has a damage over time on it that would do enough damage to kill it, other enemies still have a window to do something about it
Though ultimately I think it's okay either way, as long as it's consistent

The only ways I see it doable right now are basically redoing what I did with enemyAction (albeit simpler) and splitting the debuff loop - so I would have to duplicate enemy list and apply statuses while going through it rather then through the original - or by making debuff handling part of EnemyAction's execution (which would however also change when debuffs are handled)

Reply:
other enemies don't "still have a window to do something about it" because turn start applications are automatic things, enemies can't do something when the turn already started to mitigate those
i think we should keep it as is, because it will be consistent no matter the order of the enemies on screen and in the list

Reply:
What I meant was if we resolved those things at the start of enemy action, there could be enemy that would have it's turn before affected enemy which could cleanse it / heal it / whatever.

Reply:
ah yes I see
we will keep it as is anyway I think, because it's more consistent that way